base/timing.jl

```diff
@@ -207,6 +207,12 @@ macro time(ex)
         local stats = gc_num()
         local compile_elapsedtime = cumulative_compile_time_ns_before()
         local elapsedtime = time_ns()
+        ## ensure time samplers are compiled
+        compile_elapsedtime = cumulative_compile_time_ns_after() - compile_elapsedtime
```

Comment:
Would it be enough just to call `cumulative_compile_time_ns_after`? Looks like that is the only thing that might not have been compiled yet.

Reply:
Good point. That does fix it. 
I went with `precompile` instead of just calling them because `cumulative_compile_time_ns_after` disables timing on the thread, which I worried could complicate nested `@time`'s

This is after the latest commit
```
julia> @time map(x -> 2x, 1:3);
  0.054205 seconds (48.70 k allocations: 2.691 MiB, 99.65% compilation time)
```

Reply:
With the other ordering, is precompiling even necessary?

Reply:
You're right, I just tested and it's not. Removed!